mono/mini/interp/interp.c

```diff
@@ -3269,6 +3269,14 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 		sp++;
 	}
 
+	context->nframes ++;
+	if (G_UNLIKELY (mini_get_debug_options ()->interp_max_frames)) {
+		if (context->nframes == mini_get_debug_options ()->interp_max_frames) {
+			mini_get_debug_options ()->interp_max_frames = 0;
```

Comment:
Is there a particular reason why you clear this after the first time it activates? It feels unnecessary and in the context of wasm (where you could be calling in from outside even after a throw has happened) it means you'd have no way to re-enable the recursion limit after the first time it's hit.

Also, a stack overflow exception can't be caught unless it's thrown by "user code", IIRC - will this behave like a user throw or a runtime throw? If the former that would mean the executing code could unintentionally trap the stack overflow and clear the flag. I'm guessing this is a runtime throw, in which case this should reliably throw and abort when the limit is first hit.